postgres-types/src/pg_lsn.rs

```diff
@@ -0,0 +1,69 @@
+//! Log Sequence Number (LSN) type for PostgreSQL Write-Ahead Log
+//! (WAL), also known as the transaction log.
+
+use bytes::BytesMut;
+use postgres_protocol::types;
+use std::error::Error;
+use std::fmt;
+
+use crate::{FromSql, IsNull, ToSql, Type};
+
+/// Postgres `PG_LSN` type.
+#[derive(Clone, Copy, Eq, Ord, PartialEq, PartialOrd)]
+pub struct PgLsn(u64);
+
+impl From<u64> for PgLsn {
+    fn from(lsn_u64: u64) -> Self {
+        PgLsn(lsn_u64)
+    }
+}
+
+impl From<PgLsn> for u64 {
+    fn from(lsn: PgLsn) -> u64 {
+        lsn.0
+    }
+}
+
+impl From<&str> for PgLsn {
+    fn from(lsn_str: &str) -> Self {
+        let split: Vec<&str> = lsn_str.split('/').collect();
+        assert_eq!(split.len(), 2);
+        let (hi, lo) = (
+            u64::from_str_radix(split[0], 16).unwrap(),
+            u64::from_str_radix(split[1], 16).unwrap(),
```

Comment:
What do these two parts represent? Should there be accessors for them on the type?

Reply:
There isn't much practical meaning there. WAL files are normally 16MB, so the low 24 bits are normally the offset within a WAL file. But that's not guaranteed at a protocol level, and a compile-time switch can change the WAL file size.

I think there was a historical idea of a "log file #" (first 4 bytes) and an "offset" (last 4 bytes), but those never really made a lot of sense because they didn't correspond to the WAL file size, so you always needed both to determine the file name.

Reply:
Ah cool, so this is just there to match how Postgres renders the value.